### ISSUE TITLE
no-useless-concat rule

### ISSUE THREAD

**Issue by Cellule** [eslint/eslint#3506]
Prefer-Template doesn't detect adding 2 template string
If you have the code
```js
const a = `some` + `string`;
```
It won't be detected by the rule

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by hzoo** [comment]
Added this check in jscs as well (http://jscs.info/rule/requireTemplateStrings.html), so :+1: from me. Can do a PR

**Comment by hzoo** [comment]
And what about ``` `a` + 'b' ``` (a template literal and string literal)?

**Comment by mysticatea** [comment]
Hmm, if that concatenation is on single line, it's useless, so I think that should be warned by a rule.
But I don't think `prefer-template` should do.

`no-useless-concat`?

**Comment by Cellule** [comment]
This sounds fair enough

**Comment by nzakas** [comment]
Before making any changes, @Cellule what is it you were expecting with the code example in your original description? Did you expect `prefer-template` to warn on that case? Something else? (Please read the notice posted right below your description - it indicates the information we need to properly triage requests.)

**Comment by Cellule** [comment]
At first I was expecting `prefer-template` to warn me about ``` `a` + `b` ``` To use a template string instead of concatenating the 2.
However, after reading @mysticatea proposal, I believe having 2 rules would be preferable. `no-useless-cat` That would handle literal/template string concatenation (where we can correctly handle multiline case) and `prefer-template` to warn when adding a var to a string or another template string.

**Comment by nzakas** [comment]
Okay thanks, updated the title to reflect the change in approach for this issue.

**Comment by hzoo** [comment]
Ok I'l try a PR

**Comment by mysticatea** [comment]
@hzoo Thank you! 
Yes, if the concatenation is on multiline, it's considered it's needed for readability. I thought this check is needed for all pattern.
